framework/DataObjects/DataSet.py

```diff
@@ -657,6 +663,18 @@ def __len__(self):
     """
     return self.size
 
+  @property
+  def isEmpty(self):
+    """
+      @ In, None
+      @ Out, boolean, True if the dataset is empty otherwise False
+    """
+    try:
+      empty = False if len(self.asDataset().variables) > 0 else True
```

Comment:
calling asDataset() is not particularly cheap.  Preference should be given to checking the size of the collector and data.

Reply:
agree

Reply:
changed